File: field/src/test/java/field/FieldsTest.java
package field;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.Is.is;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
class FieldsTest {

    @Mock private Field field1;
    @Mock private Field field2;

    @Test
    void createSqlPart() {
        when(field1.createSqlPart()).thenReturn("hahah1");
        when(field2.createSqlPart()).thenReturn("hahah2");
        Fields fields = new Fields();
        fields.addField(field1).addField(field2);

        String expected = "hahah1\nhahah2";
        String actual = fields.createSqlPart();

        assertThat(actual, is(expected));
    }
}

File: kotlin-version/src/test/kotlin/json/schema/hive/FieldsTest.kt
package json.schema.hive

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.Is.`is`
import org.junit.jupiter.api.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

private const val SQL_1 = "field1 string"
private const val SQL_2 = "field2 string"

internal class FieldsTest {

    @Test
    fun createSqlPart() {
        val field1 = mock(Field::class.java)
        val field2 = mock(Field::class.java)
        `when`(field1.createSqlPart()).thenReturn(SQL_1)
        `when`(field2.createSqlPart()).thenReturn(SQL_2)

        val actual = Fields().addField(field1).addField(field2).createSqlPart()

        assertThat(actual, `is`("field1 string\nfield2 string"))
    }
}
